io/input/inputformat_factory_unittest.cpp

```diff
@@ -0,0 +1,174 @@
+#include "io/input/inputformat_factory.hpp"
+
+#include "base/session_local.hpp"
+
+#include "gtest/gtest.h"
+
+namespace husky {
+
+namespace io {
+
+namespace {
+
+class TestInputFormatFactory : public testing::Test {
+   public:
+    TestInputFormatFactory() {}
+    ~TestInputFormatFactory() {}
+
+   protected:
+    void SetUp() {}
+    void TearDown() {
+        husky::base::SessionLocal::get_thread_finalizers().clear();
```

Comment:
I don't think so, because PR #116 faced unittest failure serveral times continuously as well. I think we may have problems somewhere else.